resources/js/components/Common/Searchbar.vue

```diff
@@ -1,71 +1,84 @@
-<script lang="ts" setup>
+<script lang="ts" setup generic="T extends Option">
 
-import {computed, ref, defineProps} from 'vue'
-import {type School} from '@/Types/School'
+import { ref, defineProps, computed} from 'vue'
+import {type Option} from '@/Types/Option'
+import { onClickOutside } from '@vueuse/core'
 
-const isSearchFocused = ref(false)
+const target = ref()
+onClickOutside(target,()=>isFocused.value=false)
+
+const isFocused = ref(false)
 const searchQuery = ref('')
 const props = defineProps<{
-  schools: School[]
+  options: T[]
+  label : string
+  error: string
 }>()
 
-const filteredSchools = computed(() => props.schools
-  .toSorted((a, b) => (a.city + a.name).localeCompare(b.city + b.name))
-  .filter(({ name, city }) => (
-    name.toLowerCase().includes(searchQuery.value)) || city.toLowerCase().includes(searchQuery.value),
-  ))
-
-const selected = ref<School>()
-
-const emit = defineEmits(['change'])
+const selectedOption = ref<T>()
+const optionsRef = computed(()=>props.options.filter(
+  option=>{
+    return option.text.toLowerCase().includes(searchQuery.value)
+    || (option.title && option.title.toLowerCase().includes(searchQuery.value))
+  }))
 
-const onOptionClick = (school:School)=>{
-  selected.value = school
-  isSearchFocused.value = false
+const emit = defineEmits<{change:[option: T]}>()
 
-  emit('change', school.id)
+const onOptionClick = (option:T)=>{
+  selectedOption.value = option
+  isFocused.value = false
+  emit('change', option)
 }
 
 </script>
 
 <template>
-  <div :class="{'scale-y-100 max-h-80': isSearchFocused}" class="overflow-hidden duration-200 max-h-12 flex flex-col mt-2 bg-white/30 placeholder:text-gray-400 rounded-[.5rem] ring-2 ring-primary/30 ring-inset">
-    <div class="flex h-inherit items-center justify-center duration-200 rounded-[.5rem]" :class="{'ring-inset ring ring-primary':isSearchFocused}">
-      <div class="h-full items-center justify-center px-3">
-        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" class="size-5 stroke-primary/50" :class="{'stroke-primary': isSearchFocused}">
-          <path stroke-linecap="round" stroke-linejoin="round" d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z" />
-        </svg>
-      </div>
+  <label ref="target" class="block w-full text-sm font-medium leading-6 text-gray-900 duration-200">
+    {{ label }}
+    <div 
+      class="mt-2 font-medium text-sm leading-6 text-gray-900 overflow-hidden duration-200 max-h-12 flex flex-col bg-white/30 placeholder:text-gray-400 rounded-[.5rem] ring-2 ring-primary/30 ring-inset"
```

Comment:
Maybe you can use some defined rounded class. Or just update tailwind config for that.  

```suggestion
      class="mt-2 font-medium text-sm leading-6 text-gray-900 overflow-hidden duration-200 max-h-12 flex flex-col bg-white/30 placeholder:text-gray-400 rounded-sm ring-2 ring-primary/30 ring-inset"
```